akka-stream/src/main/scala/akka/stream/impl/NewLayout.scala

```diff
@@ -0,0 +1,297 @@
+/**
+ * Copyright (C) 2015-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.impl
+
+import akka.stream.impl.StreamLayout.{ AtomicModule, Module }
+import akka.stream.impl.fusing.GraphStages.MaterializedValueSource
+import akka.stream.{ Shape, OutPort, InPort, Attributes }
+
+import scala.collection.immutable
+
+object NewLayout {
```

Comment:
It would be helpful to have a least a three-word sketch of what the goal is and a one-sentence sketch of the explored way to get there.